### ISSUE TITLE
Loop In not working

### ISSUE THREAD

**Issue by marssystems** [lightninglabs/loop#253]
I get an error when tryin to do a "loop in".

admin@raspberrypi:~ $ loop in 1000000 6
Max swap fees for 1000000 sat Loop In: 89320 sat
CONTINUE SWAP? (y/n), expand fee detail (x): y
[loop] rpc error: code = Unknown desc = cannot initiate swap: rpc error: code = DeadlineExceeded desc = context deadline exceeded
admin@raspberrypi:~ $ loopd &
[1] 20913
admin@raspberrypi:~ $ 2020-07-17 11:08:32.142 [INF] LOOPD: Version: 0.6.0-beta commit=
2020-07-17 11:08:32.143 [INF] LNDC: Creating lnd connection to localhost:10009
2020-07-17 11:08:32.155 [INF] LNDC: Connected to lnd
unable to obtain macaroons: open /home/admin/.lnd/data/chain/bitcoin/mainnet/invoices.macaroon: no such file or directory

[1]+ Done loopd

**Comment by halseth** [comment]
Seeing similar error, but doesn't seem to have anything to do with macaroons:

loop:
```
loop in --amt 2200000
Max swap fees for 2200000 sat Loop In: 31285 sat
CONTINUE SWAP? (y/n), expand fee detail (x): y
[loop] rpc error: code = Unknown desc = cannot initiate swap: rpc error: code = DeadlineExceeded desc = context deadline exceeded
```
loopd:
```
2020-07-17 15:59:40.448 [INF] LOOPD: Loop in quote request received
2020-07-17 15:59:42.845 [INF] LOOPD: Loop in request received
2020-07-17 15:59:42.846 [INF] LOOP: Loop in 0.022 BTC (last hop: <nil>)
2020-07-17 15:59:43.118 [INF] LOOP: Initiating swap request at height 639630
2020-07-17 16:00:13.120 [ERR] LOOPD: Loop in: cannot initiate swap: rpc error: code = DeadlineExceeded desc = context deadline exceeded
```

**Comment by joostjager** [comment]
I think the client-side rpc timeout is just too short to see the server error that the probe failed.

**Comment by marssystems** [comment]
So what is the fix/workaround for this? Is there a file/setting I can edit to extend  the timeout period or is it more complicated than that?

**Comment by marssystems** [comment]
Could this be the problem?

admin@raspberrypi:~ $ loopd
2020-07-18 11:07:05.750 [INF] LOOPD: Version: 0.6.0-beta commit=
2020-07-18 11:07:05.750 [INF] LNDC: Creating lnd connection to localhost:10009
2020-07-18 11:07:05.770 [INF] LNDC: Connected to lnd
unable to obtain macaroons: open /home/admin/.lnd/data/chain/bitcoin/mainnet/invoices.macaroon: no such file or directory
admin@raspberrypi:~ $

**Comment by marssystems** [comment]
I think the macaroon file name is wrong - it should be invoice.macaroon

**Comment by marssystems** [comment]
Anyone got a fix for this?

**Comment by marssystems** [comment]
OK - I upgraded to v0.6.5-beta and I'm still getting the same error -

admin@raspberrypi:~/config.scripts $ loopd
2020-07-21 00:23:17.431 [INF] LOOPD: Version: 0.6.5-beta commit=
2020-07-21 00:23:17.432 [INF] LNDC: Creating lnd connection to localhost:10009
2020-07-21 00:23:17.433 [INF] LNDC: Connected to lnd
2020-07-21 00:23:17.531 [INF] LNDC: lnd version: v0.10.3-beta, build tags 'autopilotrpc,signrpc,walletrpc,chainrpc,invoicesrpc,watchtowerrpc'
loopd exited with an error: unable to obtain macaroons: open /home/admin/.lnd/data/chain/bitcoin/mainnet/invoices.macaroon: no such file or directory
admin@raspberrypi:~/config.scripts $ 

any suggestions?

**Comment by halseth** [comment]
@marssystems you need to find your `invoices.macaroon` and point loop to it. Check your lnd directory.

**Comment by marssystems** [comment]
I don't have an invoices.macaroon - its called invoice.macaroon in the lnd directory.

**Comment by guggero** [comment]
There should be a macaroon with the name `invoices.macaroon` (plural), it should be created when `lnd` is run with the correct build tags. Maybe you need to restart `lnd`? Or maybe it's located in a different directory?

**Comment by marssystems** [comment]
I am running RaspiBlitz 1.5.1 and LND 0.10.3.
I think that raspiblitz would build everything correctly.
The macaroon files are located in - /mnt/hdd/app-data/lnd/data/chain/bitcoin/mainnet/
There are 3 macaroon files - admin.macaroon, invoice.macaroon and readonly.macaroon in that directory.
Maybe LOOP is looking in the wrong directory - /home/admin/.lnd/data/.........

**Comment by halseth** [comment]
You need to recreate the macaroons and point loop to the correct dir.

See https://github.com/lightninglabs/loop#lnd and https://github.com/lightninglabs/loop#execute-a-swap

**Comment by marssystems** [comment]
Yes - I've already deleted and recreated the macaroons. LND only creates the 3 macaroons mentioned above.
If LND isn't creating the extra macaroons needed by LOOP, should I raise this issue with the RaspiBlitz team?
What other macaroons does LOOP need?

**Comment by halseth** [comment]
As mentioned in the readme, LND needs to be compiled with: `make install tags="signrpc walletrpc chainrpc invoicesrpc"`, but looking at the logs you posted, that seems to be the case.

Try deleting all macaroons, such that LND will re-create the correct ones :)

**Comment by marssystems** [comment]
OK - I will delete them again and see what gets recreated.
There are 3 macaroon files - admin.macaroon, invoice.macaroon and readonly.macaroon in the directory.

**Comment by marssystems** [comment]
Right - deleted all the macaroons and only the 3 macaroons, as mentioned above were recreated in - /mnt/hdd/app-data/lnd/data/chain/bitcoin/mainnet/

So something isn't working as it should.
Shall I pass this onto the RaspiBlitz team?

**Comment by marssystems** [comment]
admin@raspberrypi:~ $ loopd
2020-07-21 15:44:16.910 [INF] LOOPD: Version: 0.6.5-beta commit=
2020-07-21 15:44:16.910 [INF] LNDC: Creating lnd connection to localhost:10009
2020-07-21 15:44:16.912 [INF] LNDC: Connected to lnd
2020-07-21 15:44:17.145 [INF] LNDC: lnd version: v0.10.3-beta, build tags 'autopilotrpc,signrpc,walletrpc,chainrpc,invoicesrpc,watchtowerrpc'
loopd exited with an error: unable to obtain macaroons: open /home/admin/.lnd/data/chain/bitcoin/mainnet/invoices.macaroon: no such file or directory
admin@raspberrypi:~ $

**Comment by marssystems** [comment]
OK - passed to the RaspiBlitz support team. Should be fixed in v1.6.
